services/notification/__init__.py

```diff
@@ -101,39 +116,52 @@ def get_notifiers_instances(self) -> Iterator[AbstractBaseNotifier]:
 
         current_flags = [rf.flag_name for rf in self.repository.flags if not rf.deleted]
         for key, title, status_config in self.get_statuses(current_flags):
-            status_notifier_class = get_status_notifier_class(key, "status")
-            if self._should_use_checks_notifier():
-                checks_notifier = get_status_notifier_class(key, "checks")
-                yield ChecksWithFallback(
-                    checks_notifier=checks_notifier(
-                        repository=self.repository,
-                        title=title,
-                        notifier_yaml_settings=status_config,
-                        notifier_site_settings={},
-                        current_yaml=self.current_yaml,
-                        decoration_type=self.decoration_type,
-                        gh_installation_name_to_use=self.gh_installation_name_to_use,
-                    ),
-                    status_notifier=status_notifier_class(
+            should_use_status_notifiers = self._should_use_status_notifier(
+                status_type=key
+            )
+            should_use_checks_notifier = self._should_use_checks_notifier(
+                status_type=key
+            )
+
+            # Introduced to gate team plan not having statuses nor checks based on conditionals.
+            if not should_use_status_notifiers and not should_use_checks_notifier:
```

Comment:
i'm wondering if this is meant to skip the comment and slack notifiers, if not, or if they will be skipped anyways, then we can remove this if statement

Reply:
We should have the comment, clarifying about the slack one, but how would removing this help?

To clarify the logic here, everything remains the same, the only change introduced by this ticket is "if you're in a team plan, and you have a "changes" or "project" status checks, we don't show them, we only show "patch" specific comments" (the ticket was originally "skip everything" but its now "skip some things". There was a fn `_should_use_checks_notifier` that has some logic in it already that decides whether you show checks or not - I added a small case inside to do the team plan changes. We also always sent statuses, which changes with this ticket, so I added a fn `_should_use_status_notifier` which just adds the team plan related logic to it, otherwise returns true. 

So I don't see a way to get rid of this if ^ here.